odbc/src/api/data.rs

```diff
@@ -115,6 +115,22 @@ impl IntoCData for Bson {
             Bson::Int64(i) => Ok(i.to_le_bytes().to_vec()),
             Bson::Binary(b) => Ok(b.bytes),
             Bson::Decimal128(d) => Ok(d.bytes().to_vec()),
+            Bson::Document(d) => Ok(d.to_string().into_bytes()),
+            Bson::Array(a) => Ok(serde_json::to_string(
+                &a.into_iter().map(|b| b.to_json()).collect::<Vec<String>>(),
+            )
+            .unwrap()
+            .into_bytes()),
+            Bson::ObjectId(o) => Ok(o.to_string().into_bytes()),
+            Bson::Undefined => Ok(char::from(0).to_string().into_bytes()),
+            Bson::Null => Ok(char::from(0).to_string().into_bytes()),
+            Bson::RegularExpression(r) => Ok(r.to_string().into_bytes()),
+            Bson::DbPointer(_) => Ok(char::from(0).to_string().into_bytes()),
+            Bson::JavaScriptCode(j) => Ok(j.into_bytes()),
+            Bson::JavaScriptCodeWithScope(j) => Ok(j.to_string().into_bytes()),
+            Bson::Symbol(s) => Ok(s.into_bytes()),
+            Bson::MinKey => Ok("MinKey".to_string().into_bytes()),
+            Bson::MaxKey => Ok("MaxKey".to_string().into_bytes()),
             o => Err(ODBCError::RestrictedDataType(o.to_type_str(), BINARY)),
```

Comment:
Looks like we're missing the [TIMESTAMP](https://github.com/mongodb/mongo-odbc-driver/blob/6073ee840d275fd96a0eba229a60f0c642d87faf/core/src/bson_type_info.rs#L317) value.